File: docs/current_docs/api/snippets/debugging/python/terminal-directory-2/main.py
from dagger import dag, function, object_type


@object_type
class MyModule:
    @function
    async def advanced_directory(self) -> str:
        return await (
            dag.git("https://github.com/dagger/dagger.git")
            .head()
            .tree()
            .terminal(
                cmd=["/bin/bash"],
                experimentalprivilegednesting=false,
                insecurerootcapabilities=false,
            )
            .file("README.md")
            .contents()
        )

File: docs/current_docs/api/snippets/debugging/go/terminal-directory-2/main.go
package main

import (
	"context"
	"dagger/my-module/internal/dagger"
)

type MyModule struct{}

func (m *MyModule) AdvancedDirectory(ctx context.Context) (string, error) {
	return dag.
		Git("https://github.com/dagger/dagger.git").
		Head().
		Tree().
		Terminal(dagger.ContainerTerminalOpts{
			Cmd:                           []string{"/bin/bash"},
			ExperimentalPrivilegedNesting: false,
			InsecureRootCapabilities:      false,
		}).
		File("README.md").
		Contents(ctx)
}
